src/content/reference/react/use-client.md

```diff
@@ -23,37 +23,364 @@ canary: true
 
 ### `'use client'` {/*use-client*/}
 
-Add `'use client';` at the very top of a file to mark that the file (including any child components it uses) executes on the client, regardless of where it's imported.
+Add `'use client'` at the top of a file to mark the module and its transitive dependencies as client code.
 
-```js
+```js {1}
 'use client';
 
 import { useState } from 'react';
+import { formatDate } from './formatters';
+import Link from './link';
 
 export default function RichTextEditor(props) {
+  formatDate();
   // ...
+  return
+    <Link />
+    // ...
+}
 ```
 
-When a file marked `'use client'` is imported from a server component, [compatible bundlers](/learn/start-a-new-react-project#bleeding-edge-react-frameworks) will treat the import as the "cut-off point" between server-only code and client code. Components at or below this point in the module graph can use client-only React features like [`useState`](/reference/react/useState).
+When a file marked with `'use client'` is imported from a Server Component, [compatible bundlers](/learn/start-a-new-react-project#bleeding-edge-react-frameworks) will treat the module import as a boundary between server-run and client-run code.
+
+As a dependency of `RichTextEditor`,  `formatDate` and `Link` will also be evaluated on the client regardless if their modules contain a `'use client'` directive.
```

Comment:
regardless if -> regardless of whether

Possible follow-on sentence: Note that the same module may be evaluated on the server when imported from server code and on the client when imported from client code.